python-package/xgboost/sklearn.py

```diff
@@ -898,13 +937,30 @@ def get_params(self, deep: bool = True) -> Dict[str, Any]:
         """Get parameters."""
         # Based on: https://stackoverflow.com/questions/59248211
         # The basic flow in `get_params` is:
-        # 0. Return parameters in subclass first, by using inspect.
-        # 1. Return parameters in `XGBModel` (the base class).
+        # 0. Return parameters in subclass (self.__class__) first, by using inspect.
+        # 1. Return parameters in all parent classes (especially `XGBModel`).
         # 2. Return whatever in `**kwargs`.
         # 3. Merge them.
+        #
+        # This needs to accommodate being called recursively in the following
```

Comment:
Could you please help add a test for this? The hierarchy and the Python introspection are getting a bit confusing now. ;-(

Reply:
Sure! I just added one in https://github.com/dmlc/xgboost/pull/11021/commits/a51184888a7cf1e80c9b262a5af2f2518ca53ebc, let me know if there are other conditions you'd like to see tested.

Between that and the existing test:

https://github.com/dmlc/xgboost/blob/5826b02d036deae8c232fb35654eb21f57490919/tests/python/test_with_sklearn.py#L758

I think this behavior should be well-covered.